src/main/java/gov/va/viers/cdi/emis/ws/endpoint/MilitaryInfoEndpoint.java

```diff
@@ -30,17 +42,129 @@
+
+    try {
+      if (!request.getEdipiORicn().getInputType().equals("EDIPI")) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-03", "INVALID_IDENTIFIER");
+      } else if (request.getEdipiORicn().getEdipiORicnValue().isEmpty()) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-02", "MISSING_EDIPI");
+      }
+    } catch (DatatypeConfigurationException e) {
+      LOGGER.error("error while generating essError", e);
+    }
+
+    /* This should probably be wrapped in some null checks, not sure what EMIS does in those cases
     though with regards to returning that there was bad input*/
     JAXBElement<gov.va.schema.emis.vdrdodadapter.v2.EMISmilitaryServiceEligibilityResponseType>
         dodResponse =
             dodClient.getMilitaryServiceEligibilityResponse(
                 request.getEdipiORicn().getEdipiORicnValue());
 
+    try {
+      if (dodResponse.getValue().getESSError() != null) {
+        if (dodResponse.getValue().getESSError().getESSResponseCode().equals("ERROR")
+            && dodResponse.getValue().getESSError().getEssText().equals("INVALID_EDIPI_INPUT")) {
+          EMISMapperImpl emisMapper = new EMISMapperImpl();
+          ESSErrorType essErrorType =
+              emisMapper.mapESSErrorType(dodResponse.getValue().getESSError());
+          return generateBadFormatEdipiEssError(essErrorType, requestSoapHeaders);
+        }
+      }
+    } catch (DatatypeConfigurationException e) {
+      LOGGER.error("error when generating badFormatEdipiEssError", e);
+    }
+
     EMISmilitaryServiceEligibilityResponseType noJaxbResponse;
-    noJaxbResponse = EMISMapper.INSTANCE.mapEMISmilitaryServiceEligibilityResponseType(dodResponse.getValue());
+    noJaxbResponse =
+        EMISMapper.INSTANCE.mapEMISmilitaryServiceEligibilityResponseType(dodResponse.getValue());
 
     return objectFactory.createEMISmilitaryServiceEligibilityResponse(noJaxbResponse);
   }
+
+  private InputHeaderInfo getInputHeaderInfo(
+      SoapHeaderElement soapHeaderElement, InputHeaderInfo requestSoapHeaders) {
+    try {
+      // create unmarshaller
+      JAXBContext context =
+          JAXBContext.newInstance(gov.va.viers.cdi.cdi.commonservice.v2.ObjectFactory.class);
+      Unmarshaller unmarshaller = context.createUnmarshaller();
+
+      // unmarshall header
+      JAXBElement<InputHeaderInfo> headers =
+          (JAXBElement<InputHeaderInfo>) unmarshaller.unmarshal(soapHeaderElement.getSource());
+
+      // get header values
+      requestSoapHeaders = headers.getValue();
+
+    } catch (JAXBException e) {
+      LOGGER.error("error during unmarshalling of soap header", e);
+    }
+    return requestSoapHeaders;
+  }
+
+  private JAXBElement<EMISmilitaryServiceEligibilityResponseType> generateBadFormatEdipiEssError(
+      ESSErrorType essErrorType, InputHeaderInfo inputHeaderInfo)
```

Comment:
Passing in essErrorType is not necessary, Instead create the object in the method like in  generateMissingOrInvalidEdipiEssError()